packages/core-transaction-pool/src/memory.ts

```diff
@@ -18,6 +18,13 @@ export class Memory {
     private byId: { [key: string]: MemoryTransaction } = {};
     private bySender: { [key: string]: Set<MemoryTransaction> } = {};
     private byType: { [key: number]: Set<MemoryTransaction> } = {};
+    /**
+     * An array of transactions, sorted by expiration (lower height comes first).
+     * This array may not contain all transactions that are in the pool,
+     * transactions that are without expiration are not included. Used to:
+     * - find all transactions that have expired (have an expiration height
+     *   lower than the current height) - they are at the beginning of the array.
+     */
     private byExpiration: MemoryTransaction[] = [];
```

Comment:
Why do you need that long of a comment to understand what an array with the name `byExpiration` holds? The name is pretty self-explanatory and the code that uses it is also very simple.

If you feel the need to write that long comments, or even comments at all for such trivial things, you should think about why you really need those comments and remove the need for them and figure out how to make the code easier to understand.

Reply:
The comment is needed to convey the following information, which cannot be deduced from the name of the variable:
* The array does not contain all transactions
* Is it sorted in ascending or descending order

Reply:
Imo the information the comment tries to convey is a bit redundant since it can be quickly deduced by checking the relevant code. I personally would remove the comment, too.

Reply:
I changed this in a followup PR, see if you like it: https://github.com/ArkEcosystem/core/pull/2494/files#diff-ea1f8e8a5fdc70f328cde94d66d99dfeR20